couchbase-entity/src/main/java/com/kaufland/generation/model/TypeConversionMethodsGeneration.kt

```diff
@@ -4,19 +4,29 @@ import com.kaufland.util.TypeUtil
 import com.squareup.kotlinpoet.CodeBlock
 import com.squareup.kotlinpoet.FunSpec
 import com.squareup.kotlinpoet.TypeVariableName
-
-import java.util.Arrays
-
 import kaufland.com.coachbasebinderapi.PersistenceConfig
 
 class TypeConversionMethodsGeneration(val useSuspend: Boolean) {
 
     fun generate(): Collection<FunSpec> {
-
-        return Arrays.asList(
-            FunSpec.builder(READ_METHOD_NAME).addParameter("value", TypeUtil.any().copy(nullable = true)).addParameter("clazz", TypeUtil.classStar()).addTypeVariable(TypeVariableName.invoke("T")).returns(TypeVariableName.invoke("T?")).addCode(CodeBlock.builder().addStatement("val conversion = %T.${getTypeConversionMethod(useSuspend)}.get(clazz)", PersistenceConfig::class).beginControlFlow("if(conversion == null)").addStatement("return value as T").endControlFlow().addStatement("return conversion?.read(value) as T").build()).build(),
-
-            FunSpec.builder(WRITE_METHOD_NAME).addParameter("value", TypeUtil.any().copy(nullable = true)).addParameter("clazz", TypeUtil.classStar()).addTypeVariable(TypeVariableName.invoke("T")).returns(TypeVariableName.invoke("T?")).addCode(CodeBlock.builder().addStatement("val conversion = %T.${getTypeConversionMethod(useSuspend)}.get(clazz)", PersistenceConfig::class).beginControlFlow("if(conversion == null)").addStatement("return value as T").endControlFlow().addStatement("return conversion.write(value) as T").build()).build()
+        return listOf(
```

Comment:
Broke the statements to multiple lines for readability.
However, there are two relevant code changes in the two `addCode` blocks of both functions (sorry for the confusion):
I changed `return value as T` to `return value as T?`
and also `return conversion?.read(value) as T` to `return conversion?.read(value) as T?`

In both cases, the values to be cast are nullable, so a cast to non-nullable is dangerous. Also, the method returns a nullable anyway. Therefore it seemed more correct to me to not run the risk of causing null-pointers.